File: backend/common/security/src/main/java/com/veda/server/HashEncoder.java
package com.veda.server;

import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Component;

@Component
public class HashEncoder {

    private final PasswordEncoder passwordEncoder;

    public HashEncoder(PasswordEncoder passwordEncoder) {
        this.passwordEncoder = passwordEncoder;
    }

    public String encode(String raw) {
        return passwordEncoder.encode(raw);
    }

    public boolean matches(String raw, String encoded) {
        return passwordEncoder.matches(raw, encoded);
    }
}

File: src/main/kotlin/com/titsuko/server/security/HashEncoder.kt
package com.titsuko.server.security

import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Component

@Component
class HashEncoder(
    private val passwordEncoder: PasswordEncoder,
) {
    fun encode(raw: CharSequence): String? = passwordEncoder.encode(raw)

    fun matches(raw: CharSequence, encoder: String): Boolean =
        passwordEncoder.matches(raw, encoder)
}